---
title: "R Assignment BCB546"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
##Load libraries
library(dplyr)
library(ggplot2)
library(tidyverse)
library(paletteer)
##read in files 

fang <- read.table("C:/Users/kmjoh/Desktop/BCB546-Spring2021/assignments/R_Assignment/fang_et_al_genotypes.txt", header = T)

snp <- read.table("C:/Users/kmjoh/Desktop/BCB546-Spring2021/assignments/R_Assignment/snp_position.txt", sep = "\t", header = T)
```


```{r}
##Data investigation
##can see column counts and row counts in data summary window but can use the following if unsure
dim(fang)
dim(snp)

##see head and tail of dataframes
head(fang)
tail(fang)
head(snp)
tail(snp)

##see class of each column
sapply(fang, class)
sapply(snp, class)
```


```{r}
##subset into teosinte and maize files
teosinte <- subset.data.frame(fang, fang$Group == "ZMPBA"| fang$Group == "ZMPIL"|fang$Group == "ZMPJA")
maize <- subset.data.frame(fang, fang$Group == "ZMMIL"| fang$Group == "ZMMLR"|fang$Group == "ZMMMR")

##Remove Group and JG_OTU columns before transposition
teosinte$JG_OTU <- NULL
teosinte$Group <- NULL
maize$JG_OTU <- NULL
maize$Group <- NULL

##transpose maize and teosinte files 
maize_tr <- data.frame(t(maize))
teosinte_tr <- data.frame(t(teosinte))

##make SNP_IDs rownames in snp
row.names(snp) <- snp$SNP_ID
snp$SNP_ID <- NULL

##merge teosinte/mmaize dataframes with snps df
maize_merge <- merge(snp, maize_tr, by = 0)
teosinte_merge <- merge(snp, teosinte_tr, by = 0)

##pull out multiple and unknown into own df 
unknown_m <- subset.data.frame(maize_merge, maize_merge$Chromosome == "unknown")
unknown_t <- subset.data.frame(teosinte_merge, teosinte_merge$Chromosome == "unknown")
multiple_m <- subset.data.frame(maize_merge, maize_merge$Chromosome == "multiple")
multiple_t <- subset.data.frame(teosinte_merge, teosinte_merge$Chromosome == "multiple")


##make position and chr columns numeric so they sort correctly
teosinte_merge$Chromosome <- as.numeric(teosinte_merge$Chromosome)
teosinte_merge$Position <- as.numeric(teosinte_merge$Position)
maize_merge$Chromosome <- as.numeric(maize_merge$Chromosome)
maize_merge$Position <- as.numeric(maize_merge$Position)

##sort files by chromosome and Position
maize_asc <- maize_merge[order(maize_merge$Chromosome, maize_merge$Position),]
teosinte_asc <- teosinte_merge[order(teosinte_merge$Chromosome, teosinte_merge$Position),]

##remove NA calls 
maize_asc <- maize_asc %>% filter(!is.na(maize_asc$Chromosome))
teosinte_asc <- teosinte_asc %>% filter(!is.na(teosinte_asc$Chromosome))



##Save into files

##Create naming vector 
mnames <- c("maize1","maize2","maize3","maize4","maize5","maize6","maize7","maize8","maize9","maize10")
tnames <- c("teosinte1","teosinte2","teosinte3","teosinte4","teosinte5","teosinte6","teosinte7","teosinte8","teosinte9","teosinte10")

##Save into files subsetting by chromosome number
for (i in 1:length(maize_asc$Chromosome)) {
  write.table(subset.data.frame(maize_asc, maize_asc$Chromosome == i),
    file = paste0("asc",mnames[i], ".txt"), sep = "\t", row.names = F, quote = F)
  
}

for (i in 1:length(teosinte_asc$Chromosome)) {
  write.table(subset.data.frame(teosinte_asc, teosinte_asc$Chromosome == i),
    file = paste("asc",tnames[i], ".txt"), sep = "\t", row.names = F, quote = F)
  
}


##Create descending snp size file
maize_desc <- maize_merge[order(maize_merge$Chromosome, -maize_merge$Position),]
teosinte_desc <- teosinte_merge[order(teosinte_merge$Chromosome, -teosinte_merge$Position),]

##Replace "?" with "-" in SNP information
maize_desc <- as_tibble(lapply(maize_desc, gsub, pattern="?", replacement="-", fixed=T))
teosinte_desc <- as_tibble(lapply(teosinte_desc, gsub, pattern="?", replacement= "-", fixed=T))

##Save into files subsetting by chromosome number
for (i in 1:length(maize_desc$Chromosome)) {
  write.table(subset.data.frame(maize_desc, maize_desc$Chromosome == i),
    file = paste0("desc", mnames[i], ".txt"), sep = "\t", row.names = F, quote = F)
  
}

for (i in 1:length(teosinte_desc$Chromosome)) {
  write.table(subset.data.frame(teosinte_desc, teosinte_desc$Chromosome == i),
    file = paste("desc", tnames[i], ".txt"), sep = "\t", row.names = F, quote = F)
  
}

```


Time for visualization
```{r}
##Reshape data for visualization

t_long <- teosinte_merge %>% pivot_longer(
  cols = starts_with("X"),
  names_to = NULL,
  names_prefix = "X",
  values_to = "SNP",
)
m_long <- maize_merge %>% pivot_longer(
  cols = starts_with("X"),
  names_to = NULL,
  names_prefix = "X",
  values_to = "SNP",
)
##Change Chromosomes to factors
maize_merge$Chromosome <- factor(maize_merge$Chromosome, levels = c(1,2,3,4,5,6,7,8,9,10))
teosinte_merge$Chromosome <- factor(teosinte_merge$Chromosome, levels = c(1,2,3,4,5,6,7,8,9,10))

##Split SNP column 
t_split <- t_long %>% separate(
  SNP, c("SNP1", "SNP2"), remove = F)
m_split <- m_long %>% separate(
  SNP, c("SNP1","SNP2"), remove = F)
##Replace blanks with "missing"

##Class as heterozygote y/n based on if the two snp columns are the same

t_split$hetero <- ifelse(t_split$SNP1 == t_split$SNP2, "N", "Y")
m_split$hetero <- ifelse(m_split$SNP1 == m_split$SNP2, "N", "Y")

##Replace missing values 

m_split <- m_split %>% mutate(hetero = replace(hetero, SNP1 == "", "missing"))
t_split <- t_split %>% mutate(hetero = replace(hetero, SNP1 == "", "missing"))


##SNP counts by chromosome, using the not elongated version to give a more accurate count of snp locations
m_snp_count <- ggplot(maize_merge) + geom_bar(aes(x = Chromosome, fill = Chromosome)) + theme_minimal(base_size = 15) + scale_fill_brewer(palette = "PRGn")
ggsave("Maize_SNP_location_bars.png", plot = m_snp_count)
t_snp_count <- ggplot(teosinte_merge) + geom_bar(aes(x = Chromosome, fill = Chromosome)) + theme_minimal(base_size = 15) +scale_fill_brewer(palette = "Spectral")
ggsave("Teosinte_SNP_location_bars.png", plot = m_snp_count)

##Density plots
maize_all_density <- ggplot(maize_merge) + geom_density(aes(x=Position, fill = Chromosome), alpha = .25,) + theme_minimal()
ggsave("Maize_SNP_density.png", plot = maize_all_density)
teosinte_all_density <- ggplot(teosinte_merge) + geom_density(aes(x=Position, fill = Chromosome), alpha = .25,) + theme_minimal()
ggsave("teosinte_SNP_density.png", plot = teosinte_all_density)

##Density plots as grid
maize_all_density <- ggplot(maize_merge) + geom_density(aes(x=Position, fill = Chromosome), alpha = .25,) + theme_minimal() + facet_wrap(~Chromosome)
ggsave("Maize_SNP_density_faceted.png", plot = maize_all_density)
teosinte_all_density <- ggplot(teosinte_merge) + geom_density(aes(x=Position, fill = Chromosome), alpha = .25,) + theme_minimal() + facet_wrap(~Chromosome)
ggsave("teosinte_SNP_density_faceted.png", plot = teosinte_all_density)

##Missing data and heterozygosity plots
maize_het <- ggplot(m_split, aes(x=hetero, fill = Chromosome)) + geom_bar(position = position_dodge()) + theme_minimal() + scale_fill_brewer(palette = "Spectral") 
ggsave("Maize_heterozygosity.png", plot = maize_het)
maize_het_normalized <- ggplot(m_split, aes(x=hetero, fill = Chromosome)) + geom_bar(position = position_fill()) + theme_minimal() + scale_fill_brewer(palette = "Spectral") 
ggsave("Maize_normalized_heterozygosity.png", plot = maize_het_normalized)

teosinte_het <- ggplot(t_split, aes(x=hetero, fill = Chromosome)) + geom_bar(position = position_dodge()) + theme_minimal() + scale_fill_brewer(palette = "Spectral") 
ggsave("teosinte_heterozygosity.png", plot = teosinte_het)
teosinte_het_normalized <- ggplot(t_split, aes(x=hetero, fill = Chromosome)) + geom_bar(position = position_fill()) + theme_minimal() + scale_fill_brewer(palette = "Spectral") 
ggsave("teosinte_normalized_heterozygosity.png", plot = teosinte_het_normalized)

##Dot plot of position on Chromosomes
dots_maize <- ggplot(maize_merge, aes(x=Position, y=Chromosome, fill= Chromosome)) + geom_point(size=2, alpha=.5) + scale_fill_brewer(palette = "Paired")
ggsave("Position_along_Chromosomes_dot_plot_maize.png", plot = dots_maize)

dots_teosinte <- ggplot(teosinte_merge, aes(x=Position, y=Chromosome, fill= Chromosome)) + geom_point(size=2, alpha=.5) + scale_fill_brewer(palette = "Paired")
ggsave("Position_along_Chromosomes_dot_plot_teosinte.png", plot = dots_teosinte)
```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Ctrl+Alt+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Ctrl+Shift+K* to preview the HTML file).

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.
